### ISSUE TITLE
Odd result of splitAtParams

### ISSUE THREAD

**Issue by uebayasi** [diagrams/geometry#7]
`splitAtParams` returns odd results while I get sane ones from `splitAtParam`.

By splitting `FCubic (P (V2 0 0)) (P (V2 0 1)) (P (V2 1 0)) (P (V2 1 1))` at `0.5`, the result must be:

- `FCubic (P (V2 0.0 0.0)) (P (V2 0.0 0.5)) (P (V2 0.25 0.5)) (P (V2 0.5 0.5))`
- `FCubic (P (V2 0.5 0.5)) (P (V2 0.75 0.5)) (P (V2 1.0 0.5)) (P (V2 1.0 1.0))`

But `splitAtParams` oddly returns:

- `FCubic (P (V2 0.0 0.0)) (P (V2 0.0 (-0.5))) (P (V2 0.25 (-1.5))) (P (V2 1.0 (-3.5)))`
- `FCubic (P (V2 1.0 (-3.5))) (P (V2 (-1.25) 2.5)) (P (V2 1.0 (-0.5))) (P (V2 1.0 1.0))]`

(I've only checked FCubic now.)

**Comment by uebayasi** [comment]
Seemingly working `splitAtParams` is like:

```
splitAtParams2 :: (InSpace v n t, Ord n, Fractional n, Sectionable t) => t -> [n] -> [t]
splitAtParams2 t = splitAtParams2Help t 0 . nub . sort . filter (\x -> x >= 0 && x < 1)

splitAtParams2Help :: (InSpace v n t, Fractional n, Sectionable t) => t -> n -> [n] -> [t]
splitAtParams2Help seg _ [] = [seg]
splitAtParams2Help seg t0 (t:ts) =
    let (left, right) = splitAtParam seg (t - t0)
    in left:splitAtParams2Help right t ts
```

**Comment by cchalmers** [comment]
Thanks for the report. Yes, it looks like my original `splitAtParams` was nonsense. Unfortunately your solution doesn't rescale properly for subsequent segments:
```
seg = FCubic (P (V2 0 0)) (P (V2 0 1)) (P (V2 1 0)) (P (V2 1 1))
did = lw 8 . mconcat . zipWith lc [blue, orange, green, purple] .
    fmap stroke $ splitAtParams2 seg [0.25, 0.5, 0.75]
```

![splitatparam2](https://user-images.githubusercontent.com/1354887/51506570-d6806b00-1de4-11e9-9834-1032ff6e5322.png)

I've fixed my original code in d52e00d82308d574c839e51cb8c3c243ec47f694 (I've tried to write a justification in the comment):

![splitatparam](https://user-images.githubusercontent.com/1354887/51506601-f6b02a00-1de4-11e9-89ee-0778c5412181.png)
